### PR TITLE
Remove the cumulative SProp flag.

### LINKED ISSUE #17108: `apply` doesn't understand `Prop` vs `SProp`
```
Axiom test : forall (P : Prop), P.
Lemma test2 : forall (P : SProp), P.
apply test.
Qed.
(* Illegal application: 
The term "test" of type "forall P : Prop, P"
cannot be applied to the term
 "forall P : SProp, P" : "SProp"
This term has type "SProp" which should be coercible to "Prop". *)
```

#### Coq Version
master

### PR DESCRIPTION
**Pull request by ppedrot**
After the redesign of unification w.r.t. SProp, this flag is now useless.

Depends on #17101.

Fixes #17108

### CODE REVIEW

**Comment by ppedrot** [comment]
@coqbot run full ci

File: doc/sphinx/addendum/sprop.rst [hunk 1]
```diff
@@ -53,18 +53,7 @@ values is impredicative, and so is :math:`\SProp`:
    Check fun (A:Type) (B:A -> SProp) => (forall x:A, B x) : SProp.
 
 In order to keep conversion tractable, cumulativity for :math:`\SProp`
-is forbidden, unless the :flag:`Cumulative StrictProp` flag is turned
-on:
-
-.. coqtop:: all
-
-   Fail Check (fun (A:SProp) => A : Type).
```
Comment (SkySkimmer) [inline doc/sphinx/addendum/sprop.rst:61]: I would keep this Fail Check

**Review by SkySkimmer — APPROVED** [review]

File: test-suite/bugs/bug_14014.v [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Inductive SFalse : SProp := .
 Definition adjust_of_sprop {A} {x y : A} (pf : x <> y) : x <> y
-  := fun e : x = y => SFalse_ind (fun _ => False) (match pf e return SFalse with end).
+  := fun e : x = y => SFalse_ind (fun _ => False) (match pf e with end).
```
[the remark this answers is not in the document]
Comment (ppedrot) [reply test-suite/bugs/bug_14014.v:3]: This is to split this check apart from the one added in #17097. This way we have a better separation of concerns.

File: doc/sphinx/addendum/sprop.rst (hunk 1 above)
Comment (ppedrot) [reply doc/sphinx/addendum/sprop.rst:61]: Will fix.

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**